### ISSUE TITLE
Add optionnal AC mode (Air Conditioning) and preset

### ISSUE THREAD

**Issue by jmcollin78** [jmcollin78/versatile_thermostat#90]
The objective is to be able to add an option that says a particular thermostat is able to do AC (and not only heating).
For that case, user should be able to add preset temperature for that mode.

**Comment by WarC0zes** [comment]
Hi,
I installed the latest version 3.3.1, on restart I received an error.
If I reconfigure a thermostat, I get an error at the end of the configuration **'unknow error**'.

here are the logs:
```
2023-07-23 06:18:08.619 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/versatile_thermostat/climate.py", line 1474, in try_window_condition
    await self.async_set_hvac_mode(HVACMode.OFF)
  File "/config/custom_components/versatile_thermostat/climate.py", line 1215, in async_set_hvac_mode
    await under.set_hvac_mode(hvac_mode) or need_control_heating
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/underlyings.py", line 183, in set_hvac_mode
    await self._cancel_cycle()
TypeError: object NoneType can't be used in 'await' expression
```

```
2023-07-23 06:41:46.882 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 148, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 242, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 297, in async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 393, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/config_flow.py", line 672, in async_step_tpi
    return await self.generic_step(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/config_flow.py", line 442, in generic_step
    return await next_step_function()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/config_flow.py", line 692, in async_step_presets
    if self._infos[CONF_AC_MODE]:
       ~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: 'ac_mode'
```

```
2023-07-23 06:42:26.246 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547834139280] object NoneType can't be used in 'await' expression
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 205, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1965, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2005, in _execute_service
    return await cast(
           ^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 235, in handle_service
    return await service.entity_service_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 848, in entity_service_call
    response_data = task.result()  # pop exception if have
                    ^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1192, in async_request_call
    return await coro
           ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 892, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/climate.py", line 1215, in async_set_hvac_mode
    await under.set_hvac_mode(hvac_mode) or need_control_heating
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/versatile_thermostat/underlyings.py", line 183, in set_hvac_mode
    await self._cancel_cycle()
TypeError: object NoneType can't be used in 'await' expression
```

I'm downgrading to 3.2.6.

**Comment by jmcollin78** [comment]
Thank you @WarC0zes for the tip. I will have a look rapidly.

**Comment by jmcollin78** [comment]
Have been fixed in 3.3.2
